### ISSUE TITLE
Upgrading from Datastore 0.6.1 to 0.6.8 causes Android app to crash

### ISSUE THREAD

**Issue by djsjr** [aws-amplify/amplify-flutter#2193]
### Description

This error occurs when a user who's device has a synced local database via datastore gets updated to the latest version of datastore. I am using CLI 9.2.1 since 10 is not compatible with amplify 0.6.1 (codegen models produced incompatible models). 

For example, if a user has synced datastore and gets upgraded from Flutter Datastore 0.6.1 to 0.6.8, their app will crash within seconds of opening every time.

If a user clears Datastore, upgrades to 0.6.8, and then syncs, there is no crash. 

If a user remans on 0.6.1 there is no crash. 

It seems obvious that there's a compatibility issue with the tables when upgrading. In the past, there was never a need to clear and re-sync between versions.

I found while testing this on my simulator I get this metadata part of the error:
```
2022-09-29 16:41:58.916 14639-14935/? E/AndroidRuntime: FATAL EXCEPTION: pool-21-thread-18
    Process: com.DOMAIN.APP_NAME, PID: XXXXX
    io.reactivex.rxjava3.exceptions.UndeliverableException: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling | DataStoreException{message=Wanted 1 metadata for item with id = 282e7103-ccf8-4aa0-bf4f-6fb72f387647, but had 0., cause=null, recoverySuggestion=This is likely a bug. please report to AWS.}
```

 as well as this `no such column: __typename` part of the error:

```
2022-09-29 16:41:58.916 14639-14936/? E/amplify:flutter:datastore: DataStoreException{message=Initial cloud sync failed for User., cause=DataStoreException{message=Error in saving the model: ModelMetadata[primaryKey =User|282e7103-ccf8-4aa0-bf4f-6fb72f387647], cause=android.database.sqlite.SQLiteException: no such column: __typename (code 1 SQLITE_ERROR): , while compiling: UPDATE `ModelMetadata` SET `id` = ?, `__typename` = ?, `_deleted` = ?, `_lastChangedAt` = ?, `_version` = ? WHERE `ModelMetadata`.`id` = ?;, recoverySuggestion=See attached exception for details.}, recoverySuggestion=Check your internet connection.}
```

I'll post the entire error here since it may help:

```
2022-09-29 16:41:58.916 14639-14935/? E/AndroidRuntime: FATAL EXCEPTION: pool-21-thread-18
    Process: com.DOMAIN.APP_NAME, PID: XXXXX
    io.reactivex.rxjava3.exceptions.UndeliverableException: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling | DataStoreException{message=Wanted 1 metadata for item with id = 282e7103-ccf8-4aa0-bf4f-6fb72f387647, but had 0., cause=null, recoverySuggestion=This is likely a bug. please report to AWS.}
        at io.reactivex.rxjava3.plugins.RxJavaPlugins.onError(RxJavaPlugins.java:367)
        at io.reactivex.rxjava3.internal.operators.single.SingleCreate$Emitter.onError(SingleCreate.java:82)
        at com.amplifyframework.datastore.syncengine.VersionRepository.lambda$null$0$VersionRepository(VersionRepository.java:66)
2022-09-29 16:41:58.916 14639-14936/? E/amplify:flutter:datastore: DataStoreException{message=Initial cloud sync failed for User., cause=DataStoreException{message=Error in saving the model: ModelMetadata[primaryKey =User|282e7103-ccf8-4aa0-bf4f-6fb72f387647], cause=android.database.sqlite.SQLiteException: no such column: __typename (code 1 SQLITE_ERROR): , while compiling: UPDATE `ModelMetadata` SET `id` = ?, `__typename` = ?, `_deleted` = ?, `_lastChangedAt` = ?, `_version` = ? WHERE `ModelMetadata`.`id` = ?;, recoverySuggestion=See attached exception for details.}, recoverySuggestion=Check your internet connection.}
2022-09-29 16:41:58.916 14639-14767/? E/amplify:aws-datastore: Failure encountered while attempting to start API sync.
    DataStoreException{message=Initial sync during DataStore initialization failed., cause=java.lang.RuntimeException: DataStoreException{message=Error in saving the model: ModelMetadata[primaryKey =User|282e7103-ccf8-4aa0-bf4f-6fb72f387647], cause=android.database.sqlite.SQLiteException: no such column: __typename (code 1 SQLITE_ERROR): , while compiling: UPDATE `ModelMetadata` SET `id` = ?, `__typename` = ?, `_deleted` = ?, `_lastChangedAt` = ?, `_version` = ? WHERE `ModelMetadata`.`id` = ?;, recoverySuggestion=See attached exception for details.}, recoverySuggestion=There is a possibility that there is a bug if this error persists. Please take a look at 
    https://github.com/aws-amplify/amplify-android/issues to see if there are any existing issues that 
    match your scenario, and file an issue with the details of the bug if there isn't.}
        at com.amplifyframework.datastore.syncengine.Orchestrator.lambda$startApiSync$3$Orchestrator(Orchestrator.java:322)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$Orchestrator$PVk58tU0K8ndPJYnH_tRmf4RGwE.subscribe(Unknown Source:2)
        at io.reactivex.rxjava3.internal.operators.completable.CompletableCreate.subscribeActual(CompletableCreate.java:40)
        at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
        at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
        at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
        at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
        at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
        at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
        at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
        at io.reactivex.rxjava3.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
        at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:614)
        at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
        at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:920)
     Caused by: java.lang.RuntimeException: DataStoreException{message=Error in saving the model: ModelMetadata[primaryKey =User|282e7103-ccf8-4aa0-bf4f-6fb72f387647], cause=android.database.sqlite.SQLiteException: no such column: __typename (code 1 SQLITE_ERROR): , while compiling: UPDATE `ModelMetadata` SET `id` = ?, `__typename` = ?, `_deleted` = ?, `_lastChangedAt` = ?, `_version` = ? WHERE `ModelMetadata`.`id` = ?;, recoverySuggestion=See attached exception for details.}
        at io.reactivex.rxjava3.internal.util.ExceptionHelper.wrapOrThrow(ExceptionHelper.java:46)
        at io.reactivex.rxjava3.internal.observers.BlockingMultiObserver.blockingGet(BlockingMultiObserver.java:94)
        at io.reactivex.rxjava3.core.Completable.blockingAwait(Completable.java:1461)
        at com.amplifyframework.datastore.syncengine.Orchestrator.lambda$startApiSync$3$Orchestrator(Orchestrator.java:318)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$Orchestrator$PVk58tU0K8ndPJYnH_tRmf4RGwE.subscribe(Unknown Source:2) 
        at io.reactivex.rxjava3.internal.operators.completable.CompletableCreate.subscribeActual(CompletableCreate.java:40) 
        at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850) 
        at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51) 
        at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850) 
        at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51) 
        at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850) 
        at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51) 
        at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850) 
        at io.reactivex.rxjava3.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64) 
        at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:614) 
        at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65) 
        at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:920) 
2022-09-29 16:41:58.916 14639-14767/? E/amplify:aws-datastore: Caused by: DataStoreException{message=Error in saving the model: ModelMetadata[primaryKey =User|282e7103-ccf8-4aa0-bf4f-6fb72f387647], cause=android.database.sqlite.SQLiteException: no such column: __typename (code 1 SQLITE_ERROR): , while compiling: UPDATE `ModelMetadata` SET `id` = ?, `__typename` = ?, `_deleted` = ?, `_lastChangedAt` = ?, `_version` = ? WHERE `ModelMetadata`.`id` = ?;, recoverySuggestion=See attached exception for details.}
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$save$3$SQLiteStorageAdapter(SQLiteStorageAdapter.java:382)
        at com.amplifyframework.datastore.storage.sqlite.-$$Lambda$SQLiteStorageAdapter$0Q4t2Se5oVsIb4pxkwWPcTH3ADs.run(Unknown Source:12)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        	... 3 more
     Caused by: android.database.sqlite.SQLiteException: no such column: __typename (code 1 SQLITE_ERROR): , while compiling: UPDATE `ModelMetadata` SET `id` = ?, `__typename` = ?, `_deleted` = ?, `_lastChangedAt` = ?, `_version` = ? WHERE `ModelMetadata`.`id` = ?;
        at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
        at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1047)
        at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:654)
        at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
        at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:62)
        at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:34)
        at android.database.sqlite.SQLiteDatabase.compileStatement(SQLiteDatabase.java:1224)
        at com.amplifyframework.datastore.storage.sqlite.SQLCommandProcessor.execute(SQLCommandProcessor.java:73)
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.writeData(SQLiteStorageAdapter.java:773)
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$save$3$SQLiteStorageAdapter(SQLiteStorageAdapter.java:365)
        	... 6 more
```

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [X] DataStore
- [ ] Storage

### Steps to Reproduce

1. Use Datastore 0.6.1
2. Create schema, add content
3. Run device that has synced datastore
4. upgrade app to 0.6.8
5. Run upgraded app on device WITHOUT uninstalling app or clearing local data on device
6. Receive error and crash a few seconds upon reopening app

### Platforms

- [ ] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Android Device/Emulator API Level

API 31, API 32+

### Environment

```bash
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.0.1 21A559 darwin-arm, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.71.0)
[✓] Connected device (5 available)
    ! Error: iPhone is busy: Preparing the watch for development via Danny’s iPhone. Xcode will continue when Danny’s iPhone is finished. (code -10)
    ! Error: iPhone is not connected. Xcode will continue when Danny’s iPhone is connected. (code -13)
[✓] HTTP Host Availability
```

### Dependencies

```bash
Dart SDK 2.17.6
Flutter SDK 3.0.5
see_app 3.2.3+113

dependencies:
- adaptive_dialog 1.8.0+1 [animations collection dynamic_color flutter intersperse macos_ui meta]
- amplify_analytics_pinpoint 0.6.8 [amplify_analytics_pinpoint_android amplify_analytics_pinpoint_ios amplify_core aws_common flutter meta]
- amplify_api 0.6.8 [amplify_api_android amplify_api_ios amplify_core amplify_flutter aws_common collection flutter meta plugin_platform_interface]
- amplify_auth_cognito 0.6.8 [amplify_auth_cognito_android amplify_auth_cognito_ios amplify_core aws_common collection flutter meta plugin_platform_interface]
- amplify_core 0.6.8 [aws_common collection flutter intl json_annotation meta plugin_platform_interface uuid]
- amplify_datastore 0.6.8 [flutter amplify_datastore_plugin_interface amplify_core plugin_platform_interface meta collection async]
- amplify_flutter 0.6.8 [amplify_core amplify_datastore_plugin_interface amplify_flutter_android amplify_flutter_ios aws_common collection flutter meta plugin_platform_interface]
- amplify_storage_s3 0.6.8 [amplify_storage_s3_android amplify_storage_s3_ios amplify_core aws_common flutter meta plugin_platform_interface path_provider path]
- async 2.8.2 [collection meta]
- audio_service 0.18.7 [audio_service_platform_interface audio_service_web audio_session rxdart flutter_cache_manager clock js flutter flutter_web_plugins]
- audio_session 0.1.10 [flutter flutter_web_plugins rxdart meta]
- back_button_interceptor 6.0.1 [collection flutter]
- badges 2.0.3 [flutter]
- chewie 1.3.5 [cupertino_icons flutter provider video_player wakelock]
- connectivity_plus 2.3.7 [flutter connectivity_plus_platform_interface connectivity_plus_linux connectivity_plus_macos connectivity_plus_web connectivity_plus_windows]
- cupertino_icons 1.0.5
- email_validator 2.1.17
- firebase_core 1.24.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_messaging 12.0.3 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_app_badger 1.5.0 [flutter]
- flutter_keyboard_visibility 5.3.0 [meta flutter_keyboard_visibility_platform_interface flutter_keyboard_visibility_web flutter]
- http 0.13.5 [async http_parser meta path]
- intl 0.17.0 [clock path]
- just_audio 0.9.29 [just_audio_platform_interface just_audio_web audio_session rxdart path path_provider async uuid crypto meta flutter]
- keyboard_dismisser 3.0.0 [flutter]
- persistent_bottom_nav_bar_v2 4.2.5 [flutter]
- provider 6.0.3 [collection flutter nested]
- readmore 2.2.0 [flutter]
- shape_of_view_null_safe 2.0.0 [flutter vector_math]
- shared_preferences 2.0.15 [flutter shared_preferences_android shared_preferences_ios shared_preferences_linux shared_preferences_macos shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- simple_rich_text 2.0.49 [flutter url_launcher]
- video_player 2.4.7 [flutter html video_player_android video_player_avfoundation video_player_platform_interface video_player_web]
- visibility_detector 0.3.3 [flutter]
- webview_flutter 3.0.4 [flutter webview_flutter_android webview_flutter_platform_interface webview_flutter_wkwebview]

transitive dependencies:
- _flutterfire_internals 1.0.1 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- amplify_analytics_pinpoint_android 0.6.8 [flutter]
- amplify_analytics_pinpoint_ios 0.6.8 [flutter]
- amplify_api_android 0.6.8 [flutter]
- amplify_api_ios 0.6.8 [amplify_core flutter]
- amplify_auth_cognito_android 0.6.8 [flutter]
- amplify_auth_cognito_ios 0.6.8 [amplify_core flutter]
- amplify_datastore_plugin_interface 0.6.8 [amplify_core collection flutter meta]
- amplify_flutter_android 0.6.8 [flutter]
- amplify_flutter_ios 0.6.8 [amplify_core flutter]
- amplify_storage_s3_android 0.6.8 [flutter]
- amplify_storage_s3_ios 0.6.8 [flutter]
- animations 2.0.5 [flutter]
- args 2.3.1
- audio_service_platform_interface 0.1.0 [flutter plugin_platform_interface meta]
- audio_service_web 0.1.1 [audio_service_platform_interface rxdart js flutter flutter_web_plugins]
- aws_common 0.1.1 [async collection http meta stream_transform uuid]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- cloud_firestore_platform_interface 5.7.6 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 2.8.9 [_flutterfire_internals cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.16.0
- connectivity_plus_linux 1.3.1 [flutter connectivity_plus_platform_interface meta nm]
- connectivity_plus_macos 1.2.4 [connectivity_plus_platform_interface flutter]
- connectivity_plus_platform_interface 1.2.1 [flutter meta plugin_platform_interface]
- connectivity_plus_web 1.2.4 [connectivity_plus_platform_interface flutter_web_plugins flutter js]
- connectivity_plus_windows 1.2.2 [connectivity_plus_platform_interface flutter]
- crypto 3.0.2 [typed_data]
- csslib 0.17.2 [source_span]
- dbus 0.7.8 [args ffi meta xml]
- dynamic_color 1.4.0 [flutter flutter_test material_color_utilities]
- fake_async 1.3.0 [clock collection]
- ffi 2.0.1
- file 6.1.4 [meta path]
- firebase_core_platform_interface 4.5.1 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 1.7.3 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_messaging_platform_interface 4.1.6 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 3.1.6 [_flutterfire_internals firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- flutter_cache_manager 3.3.0 [clock collection file flutter http path path_provider pedantic rxdart sqflite uuid]
- flutter_keyboard_visibility_platform_interface 2.0.0 [flutter meta plugin_platform_interface]
- flutter_keyboard_visibility_web 2.0.0 [flutter_keyboard_visibility_platform_interface flutter_web_plugins flutter]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- html 0.15.0 [csslib source_span]
- http_parser 4.0.1 [collection source_span string_scanner typed_data]
- intersperse 2.0.0
- js 0.6.4
- json_annotation 4.7.0 [meta]
- just_audio_platform_interface 4.2.0 [flutter plugin_platform_interface]
- just_audio_web 0.4.7 [just_audio_platform_interface flutter flutter_web_plugins]
- macos_ui 1.7.5 [flutter]
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.4
- meta 1.7.0
- nested 1.0.0 [flutter]
- nm 0.5.0 [dbus]
- path 1.8.1
- path_provider 2.0.11 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.20 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.11 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.7 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.6 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.5 [flutter platform plugin_platform_interface]
- path_provider_windows 2.1.3 [ffi flutter path path_provider_platform_interface win32]
- pedantic 1.11.1
- petitparser 5.0.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.3 [meta]
- process 4.2.4 [file path platform]
- rxdart 0.27.5
- shared_preferences_android 2.0.13 [flutter shared_preferences_platform_interface]
- shared_preferences_ios 2.1.1 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.1.1 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_macos 2.0.4 [flutter shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.1.0 [flutter plugin_platform_interface]
- shared_preferences_web 2.0.4 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.1.1 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- sky_engine 0.0.99
- source_span 1.8.2 [collection path term_glyph]
- sqflite 2.0.3+1 [flutter sqflite_common path]
- sqflite_common 2.3.0 [synchronized path meta]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- stream_transform 2.0.0
- string_scanner 1.1.0 [charcode source_span]
- synchronized 3.0.0+3
- term_glyph 1.2.0
- test_api 0.4.9 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.1 [collection]
- url_launcher 6.1.6 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- url_launcher_android 6.0.19 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.1.1 [flutter plugin_platform_interface]
- url_launcher_web 2.0.13 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.1 [flutter url_launcher_platform_interface]
- uuid 3.0.6 [crypto]
- vector_math 2.1.2
- video_player_android 2.3.9 [flutter video_player_platform_interface]
- video_player_avfoundation 2.3.6 [flutter video_player_platform_interface]
- video_player_platform_interface 5.1.4 [flutter plugin_platform_interface]
- video_player_web 2.0.12 [flutter flutter_web_plugins video_player_platform_interface]
- wakelock 0.6.2 [flutter meta wakelock_macos wakelock_platform_interface wakelock_web wakelock_windows]
- wakelock_macos 0.4.0 [flutter flutter_web_plugins wakelock_platform_interface]
- wakelock_platform_interface 0.3.0 [flutter meta]
- wakelock_web 0.4.0 [flutter flutter_web_plugins js wakelock_platform_interface]
- wakelock_windows 0.2.1 [flutter wakelock_platform_interface win32]
- webview_flutter_android 2.10.3 [flutter webview_flutter_platform_interface]
- webview_flutter_platform_interface 1.9.5 [flutter meta plugin_platform_interface]
- webview_flutter_wkwebview 2.9.5 [flutter path webview_flutter_platform_interface]
- win32 3.0.0 [ffi]
- xdg_directories 0.2.0+2 [meta path process]
- xml 6.1.0 [collection meta petitparser]
```

### Device

Pixel 4, Motorola Moto G Power (2021), other android devices used by testers

### OS

Android 11, 12

### Deployment Method

Amplify CLI

### CLI Version

9.2.1

### Amplify Config

const amplifyconfig = ''' {
    "UserAgent": "aws-amplify-cli/2.0",
    "Version": "1.0",
    "auth": {
        "plugins": {
            "awsCognitoAuthPlugin": {
                "UserAgent": "aws-amplify-cli/0.1.0",
                "Version": "0.1.0",
                "IdentityManager": {
                    "Default": {}
                },
                "CredentialsProvider": {
                    "CognitoIdentity": {
                        "Default": {
                            "PoolId": "",
                            "Region": "us-east-1"
                        }
                    }
                },
                "CognitoUserPool": {
                    "Default": {
                        "PoolId": "",
                        "AppClientId": "",
                        "Region": "us-east-1"
                    }
                },
                "Auth": {
                    "Default": {
                        "authenticationFlowType": "USER_SRP_AUTH",
                        "socialProviders": [],
                        "usernameAttributes": [
                            "EMAIL"
                        ],
                        "signupAttributes": [
                            "NAME"
                        ],
                        "passwordProtectionSettings": {
                            "passwordPolicyMinLength":,
                            "passwordPolicyCharacters": [
                            ]
                        },
                        "mfaConfiguration": "OFF",
                        "mfaTypes": [],
                        "verificationMechanisms": [
                            "EMAIL"
                        ]
                    }
                },
                "AppSync": {
                    "Default": {
                        "ApiUrl": "",
                        "Region": "us-east-1",
                        "AuthMode": "AMAZON_COGNITO_USER_POOLS",
                        "ClientDatabasePrefix": ""
                    },
                    "seeapp_AWS_IAM": {
                        "ApiUrl": "",
                        "Region": "us-east-1",
                        "AuthMode": "AWS_IAM",
                        "ClientDatabasePrefix": ""
                    }
                },
                "PinpointAnalytics": {
                    "Default": {
                        "AppId": "",
                        "Region": "us-east-1"
                    }
                },
                "PinpointTargeting": {
                    "Default": {
                        "Region": "us-east-1"
                    }
                },
                "S3TransferUtility": {
                    "Default": {
                        "Bucket": "",
                        "Region": "us-east-1"
                    }
                }
            }
        }
    },
    "storage": {
        "plugins": {
            "awsS3StoragePlugin": {
                "bucket": "",
                "region": "us-east-1",
                "defaultAccessLevel": "guest"
            }
        }
    },
    "api": {
        "plugins": {
            "awsAPIPlugin": {
                "seeapp": {
                    "endpointType": "GraphQL",
                    "endpoint": "",
                    "region": "us-east-1",
                    "authorizationType": "AMAZON_COGNITO_USER_POOLS"
                }
            }
        }
    },
    "analytics": {
        "plugins": {
            "awsPinpointAnalyticsPlugin": {
                "pinpointAnalytics": {
                    "appId": "",
                    "region": "us-east-1"
                },
                "pinpointTargeting": {
                    "region": "us-east-1"
                }
            }
        }
    }
}''';

**Comment by fjnoyp** [comment]
Hi @djsjr thanks for sharing this issue.  We'll take a look.  

Have you tested and gotten the same issue on iOS?

**Comment by djsjr** [comment]
Yes thats right. Tested on iOS with no problem.

**Comment by djsjr** [comment]
@fjnoyp any update on this?

**Comment by fjnoyp** [comment]
Hi @djsjr I think that we added support for custom primary key for datastore in between those versions.  That feature might have invalidated the existing SQL tables on your device thus causing the crash you observed. I'm checking that internally. 

In the meantime, is there a specific reason you cannot clear and redownload the app?

**Comment by djsjr** [comment]
Thanks for looking into this @fjnoyp . I have many Android beta testers - some of which began reporting crashes as soon as I upgraded amplify and put out a new version. I quickly reverted before it spread to everyone. It'd be a bit difficult to know when everyone receives the new version (Google Play store notoriously caches old versions for days even when new ones are available) and the. make sure each one then understands they must uninstall and reinstall when the auto update happens and their version starts crashing. 

Even if possible, as a developer I'd rather not have to force testers to encounter a crash in order to proceed.

**Comment by fjnoyp** [comment]
Thank you very much for the explaining your situation here.  Sounds very difficult for you to require everyone to delete and reinstall. 

I've reached out internally, and we're looking into this.  I or someone else from our team will reach out with updates once we have more information.

**Comment by fjnoyp** [comment]
Hi @djsjr 

The problem you encountered is an issue on Amplify Android tracked here: 
https://github.com/aws-amplify/amplify-android/issues/2066

A temporary workaround to avoid requiring an app uninstall:

1) Change the version string in ModelProvider dart file.  This will trigger a single table refresh. 

2) Call Datastore.clear() once when upgrading from older to newer versions of app

**Comment by Jordan-Nelson** [comment]
@djsjr - This issue should be resolved in v0.6.11. Can you please let us know if upgrading resolves your issue? Thanks.

**Comment by Jordan-Nelson** [comment]
@djsjr - I am going to close this out since we have not heard back. Please open a new issue if you are still have trouble after upgrading. Thanks.
